service/datasource-service.py

```diff
@@ -155,12 +162,52 @@ def decorated(*args, **kwargs):
 
     return decorated
 
+def generate_sheetdata(url,auth,params,headers,sheets,ids,names,direction,start,since):
+    logger.info("downloading from url=%s with params=%s" % (url, params))
+    r = requests.get(url, auth=auth, headers=headers, params=params)
+    r.raise_for_status()
+    logger.debug("Got file data")
+    workbook = xlrd.open_workbook("sesm.xsls", file_contents=r.content)
+    if workbook.props["modified"] > since:
+        iterable_sheets = sheets or range(0,Workbook.nsheets)
+        for sheet_index in iterable_sheets:
+            worksheet = workbook.sheet_by_index(sheet_index)
+            id_prefix = str(sheet_index+1) + "-" if len(iterable_sheets)>1 else ""
+            if direction == "row":
+                columnNames = getColNames(worksheet, names, start)
+                yield from getSheetRowData(worksheet, columnNames, start, ids, workbook.props["modified"], workbook.datemode, id_prefix)
+            else:
+                rowNames = getRowNames(worksheet, names, start)
+                yield from getSheetColData(worksheet, rowNames, start, ids, workbook.props["modified"], workbook.datemode, id_prefix)
 
-@app.route('/')
+@app.route('/', methods=["GET"])
+@app.route('/<path:path>', methods=["GET"])
```

Comment:
If adding a route that expects a path, I suggest not putting in on the root level. Makes it harder to implement other endpoints in the future. (This has caused issues in other microservices when suddenly more than one endpoint is needed.)